### PR TITLE
AG-13362 Bundle size reporting

### PR DESCRIPTION
**Pull request by StephenCooper**
First step to ensure we have some tests that keep track of the current module sizes.

### CODE REVIEW

File: testing/module-size/moduleValidator.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+const path = require('path');
+const resultsFilePath = path.resolve(__dirname, 'module-size-results.json');
+
+function validateSizes() {
+    console.log('Running module size tests...');
+
+    // Read and parse the module-size-results.json file
+    const results = JSON.parse(fs.readFileSync(resultsFilePath, 'utf8'));
+
+    // validate that all results their selfSize is less than the expectedSize + 2%
```
Comment (taktran) [inline testing/module-size/moduleValidator.ts:17]:
```suggestion
    // Validate that, for all results, their selfSize is less than the expectedSize + 2%
```

File: testing/module-size/vite.config.mts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+// https://vitejs.dev/config/
+export default defineConfig({
+    plugins: [react()],
+    build: {
+        rollupOptions: {
+            output: {
+                manualChunks: {
+                    react: ['react', 'react-dom'],
+                    // agGridCommunity: ['ag-grid-community'],
```
Comment (taktran) [inline testing/module-size/vite.config.mts:12]: Not needed?

**Review by taktran — APPROVED** [review]
Nice one 👍🏼 Just some minor comments

File: testing/module-size/project.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      "executor": "nx:run-commands",
+      "inputs": [],
+      "outputs": [],
+      "options": {
+        "cwd": "testing/module-size",
+        "commands": ["npm run test"]
+      }
+    }
+  },
+  "tags": ["module", "test"]
```
Comment (StephenCooper) [inline testing/module-size/project.json]: Updating these so that it is not part of the CI and will only be included in the Full build

**Pull request merged**